### PR TITLE
Prep for v1.17.0

### PR DESCRIPTION
**Pull request by odow**
## TODO

 - [x] #3567
 - [x] #3575 
 - [x] #3587
 - [x] #3590
 - [x] #3591
 - [x] #3592
 - [x] #3593
 - [x] #3595

## Pre-release

 - [x] Check that the pinned packages in `docs/Project.toml` are updated. We pin
       the versions so that changes in the solvers (changes in printing, small
       numeric changes) do not break the printing of the JuMP docs in arbitrary
       commits.
 - [x] Check that the `rev` fields in `docs/packages.toml` are updated. We pin
       the versions of solvers and extensions to ensure that changes to their
       READMEs do not break the JuMP docs in arbitrary commits, and to ensure
       that the versions are compatible with the latest JuMP and
       MathOptInterface releases.
 - [x] Update `docs/src/changelog.md`
 - [x] https://github.com/jump-dev/JuMP.jl/actions/runs/7009775299
 - [x] Change the version number in `Project.toml`
 - [x] Update the links in README.md
 - [x] The commit messages in this PR do not contain `[ci skip]`

## The release

 - [ ] After merging this pull request, comment `[at]JuliaRegistrator register` in
       the GitHub commit. This should automatically publish a new version to the
       Julia registry, as well as create a tag, and rebuild the documentation
       for this tag.

       These steps can take quite a bit of time (1 hour or more), so don't be
       surprised if the new documentation takes a while to appear. In addition,
       the links in the README will be broken until JuliaHub fetches the new
       version on their servers.

## Post-release

 - [ ] Once the tag is created, update the relevant `release-` branch. The latest
       release branch at the time of writing is `release-1.0` (we haven't
       back-ported any patches that needed to create a `release-1.Y` branch). To
       to update the release branch with the v1.10.0 tag, do:
       ```
       git checkout release-1.0
       git pull
       git merge v1.10.0
       git push
       ```

### CODE REVIEW

File: docs/src/changelog.md [hunk 1]
```diff
@@ -7,6 +7,47 @@ CurrentModule = JuMP
...
+   and [`Containers.SparseAxisArray`](@ref) (#3587)
+
+### Other
+
+ - Improved error messages for:
+    - Addition and subtraction between a matrix and a scalar (#3557) (#3558)
+    - Variables with non-constant bounds (#3583)
+    - Invalid indicator constraints (#3584)
+ - Added new solvers to the documentation:
+    - [EAGO.jl](@ref) (#3560) (#3561)
```
Comment (odow) [inline docs/src/changelog.md:37]:
```suggestion
    - `EAGO.jl` (#3560) (#3561)
```

File: docs/src/changelog.md [hunk 2]
```diff
@@ -7,6 +7,47 @@ CurrentModule = JuMP
...
+    - [Manopt.jl](@ref) (#3568)
+ - Added new tutorials:
+    - [Approximating nonlinear functions](@ref) (#3563)
+    - [Classifiers](@ref) (#3569)
+ - Improved documentation for:
+    - [`Semicontinuous`](@ref) and [`Semiinteger`](@ref) variables (#3562)
+    - [`SOS1`](@ref) and [`SOS2`](@ref) (#3565)
+    - [`start_value`](@ref) of [`MOI.HermitianPSDCone`](@ref) (#3564)
```
Comment (odow) [inline docs/src/changelog.md:45]:
```suggestion
    - [`start_value`](@ref) of [`HermitianPSDCone`](@ref) (#3564)
```

File: docs/src/changelog.md [hunk 3]
```diff
@@ -7,6 +7,50 @@ CurrentModule = JuMP
...
+ - Added new tutorials:
+    - [Approximating nonlinear functions](@ref) (#3563)
+    - [Classifiers](@ref) (#3569)
+ - Improved documentation for:
+    - [`Semicontinuous`](@ref) and [`Semiinteger`](@ref) variables (#3562)
+    - [`SOS1`](@ref) and [`SOS2`](@ref) (#3565)
+    - [`start_value`](@ref) of [`HermitianPSDCone`](@ref) (#3564)
+    - Function tracing (#3570)
+    - Nonlinear operators with vector arguments (#3577)
+    - Indicator connstraints (#3582)
```
Comment (odow) [inline docs/src/changelog.md:51]:
```suggestion
    - Indicator constraints (#3582)
```

**Comment by odow** [comment]
x-ref https://jump.dev/JuMP.jl/previews/PR3588/release_notes/

There are a couple of new things here that may have slipped past people and are worth double checking we're happy with the naming of:

 * https://jump.dev/JuMP.jl/previews/PR3588/api/JuMP/#JuMP.SkipModelConvertScalarSetWrapper
 * https://jump.dev/JuMP.jl/previews/PR3588/manual/models/#Get-the-matrix-representation

File: docs/src/changelog.md [hunk 4]
```diff
@@ -7,6 +7,50 @@ CurrentModule = JuMP
...
+   IJulia output (#3575)
+
+### Other
+
+ - Improved error messages for:
+    - Addition and subtraction between a matrix and a scalar (#3557) (#3558)
+    - Variables with non-constant bounds (#3583)
+    - Invalid indicator constraints (#3584)
+ - Added new solvers to the documentation:
+    - `EAGO.jl` (#3560) (#3561)
```
Comment (blegat) [inline docs/src/changelog.md:40]: Add Percival

File: docs/src/changelog.md [hunk 5]
```diff
@@ -7,6 +7,50 @@ CurrentModule = JuMP
...
+    - [Manopt.jl](@ref) (#3568)
```
Comment (odow) [inline docs/src/changelog.md:41]:
```suggestion
    - [Manopt.jl](@ref) (#3568)
    - `Percival.jl` (#3567)
```

**Review by blegat — APPROVED** [review]

**Comment by odow** [comment]
For `SkipModelConvertScalarSetWrapper` add note that it is mainly intended for extensions.

**Comment by odow** [comment]
For `lp_matrix_data`, offer hints for boolean and integer variables. Maybe other stuff as well

File: docs/src/changelog.md [hunk 6]
```diff
@@ -7,6 +7,52 @@ CurrentModule = JuMP
...
 
+## Version 1.17.0 (November 30, 2023)
+
+### Added
+
+ - Added [`start_value`](@ref), [`lower_bound`](@ref), and [`upper_bound`](@ref)
+   support for [`GenericAffExpr`](@ref) that are equivalent to a single
+   [`GenericVariableRef`](@ref) (#3551)
+ - Added [`SkipModelConvertScalarSetWrapper`](@ref) which is useful for
+   extensions looking to avoid [`model_convert`](@ref) (#3552)
```
Comment (odow) [inline docs/src/changelog.md:18]:
```suggestion
   extensions looking to avoid [`model_convert`](@ref) (#3552) (#3592)
```

**Pull request merged**